src/inject/dynamic-theme/stylesheet-proxy.ts

```diff
@@ -17,29 +17,37 @@ export function injectProxy() {
 
     function proxyAddRule(selector?: string, style?: string, index?: number): number {
         addRuleDescriptor.value.call(this, selector, style, index);
-        this.ownerNode.dispatchEvent(updateSheetEvent);
+        if (this.ownerNode && !this.ownerNode.classList.contains('darkreader')) {
+            this.ownerNode.dispatchEvent(updateSheetEvent);
+        }
         // Should always returns -1 https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/addRule#Return_value.
         return -1;
     }
 
     function proxyInsertRule(rule: string, index?: number): number {
         const returnValue = insertRuleDescriptor.value.call(this, rule, index);
-        this.ownerNode.dispatchEvent(updateSheetEvent);
+        if (this.ownerNode && !this.ownerNode.classList.contains('darkreader')) {
+            this.ownerNode.dispatchEvent(updateSheetEvent);
+        }
         return returnValue;
     }
 
     function proxyDeleteRule(index: number): void {
         deleteRuleDescriptor.value.call(this, index);
-        this.ownerNode.dispatchEvent(updateSheetEvent);
+        if (this.ownerNode && !this.ownerNode.classList.contains('darkreader')) {
+            this.ownerNode.dispatchEvent(updateSheetEvent);
+        }
     }
 
     function proxyRemoveRule(index?: number): void {
         removeRuleDescriptor.value.call(this, index);
-        this.ownerNode.dispatchEvent(updateSheetEvent);
+        if (this.ownerNode && !this.ownerNode.classList.contains('darkreader')) {
+            this.ownerNode.dispatchEvent(updateSheetEvent);
+        }
     }
 
-    Object.defineProperty(CSSStyleSheet.prototype, 'addRule', {...addRuleDescriptor, value: proxyAddRule});
-    Object.defineProperty(CSSStyleSheet.prototype, 'insertRule', {...insertRuleDescriptor, value: proxyInsertRule});
-    Object.defineProperty(CSSStyleSheet.prototype, 'deleteRule', {...deleteRuleDescriptor, value: proxyDeleteRule});
-    Object.defineProperty(CSSStyleSheet.prototype, 'removeRule', {...removeRuleDescriptor, value: proxyRemoveRule});
+    Object.defineProperty(CSSStyleSheet.prototype, 'addRule', Object.assign(Object.assign({}, addRuleDescriptor), {value: proxyAddRule}));
```

Comment:
Object.assign accepts many objects

Reply:
> Object.assign accepts many objects

Well yes but it also changes every references of the variable. so assigning it into a empty `{}` prevents this.

Reply:
I mean Object.assign can accept 3 and many arguments, only the first one will be affected by new properties, so don't put Object.assign inside of Object.assign

Reply:
Oh, huh. this is what the dark reader's API(ES5) has been doing.

Reply:
Then the authors of the transpiler didn't care much about it.

Reply:
So something like
```javascript
const valueRuleDescriptor = Object.assign({}, addRuleDescriptor);
Object.defineProperty(CSSStyleSheet.prototype, 'addRule', Object.assign(valueRuleDescriptor, {value: proxyAddRule}));
```
should be used?

Reply:
`Object.defineProperty(CSSStyleSheet.prototype, 'addRule', Object.assign({}, addRuleDescriptor, {value: proxyAddRule}));`